src/irr/video/COpenGLSpecializedShader.h

```diff
@@ -19,28 +20,30 @@ namespace video
 
 class COpenGLSpecializedShader : public core::impl::ResolveAlignment<IGPUSpecializedShader,core::AllocationOverrideBase<128> >
 {
+		using SMember = asset::impl::SShaderMemoryBlock::SMember;
+
+	public:
 		struct SProgramBinary {
-			GLenum format;
+			GLenum format = 0;
 			core::smart_refctd_dynamic_array<uint8_t> binary;
 		};
+		struct SUniform {
+			SUniform(const SMember& _m) : m(_m) {}
+			SMember m;
+		};
 
-	public:
 		COpenGLSpecializedShader(size_t _ctxCount, uint32_t _ctxID, uint32_t _GLSLversion, const asset::ICPUBuffer* _spirv, const asset::ISpecializedShader::SInfo& _specInfo, const asset::CIntrospectionData* _introspection);
 
-		inline GLuint getGLnameForCtx(uint32_t _ctxID) const
-		{
-			if ((*m_GLnames)[_ctxID])
-				return (*m_GLnames)[_ctxID];
+		void setUniformsImitatingPushConstants(const uint8_t* _pcData, GLuint _GLname, const GLint* _locations, uint8_t* _state) const;
```

Comment:
that can move to the pipeline since that will hold program objects now.

Reply:
not really because COpenGLSpecializedShader holds offsets of "uniforms"

Reply:
ok